index.js

```diff
@@ -45,13 +45,13 @@ class FetchHandler {
       const dataset = this.dataset.match(null, null, null, rdf.namedNode(req.iri))
 
       if (dataset.length === 0) {
-        return next()
+        return
       }
 
       const graph = rdf.graph(dataset)
 
       return res.graph(graph)
-    }).catch(next)
+    }).asCallback(next)
```

Comment:
Doesn't that always call `next`? `next` should be only called in the `dataset.length === 0` case (404). `res.graph(graph)` sends the content and closes the response stream, so following middlewares should not be processed.

Reply:
Any idea how to write a test for this situation? That would be very helpful.

Reply:
I added a test for not calling the next middleware in [this branch](https://github.com/zazuko/trifid-handler-fetch/tree/test-next-middleware), which fails with the latest changes, but works with the old code. I tried to reproduce the error, which triggered this PR, but without success. Any hints how to reproduce it? I would like to add also a test for that case.

Reply:
Nice, thanks!
> Any hints how to reproduce it? I would like to add also a test for that case.

Unfortunately not. Tried everything yesterday night including things like `const Promise = require('bluebird'); global.Promise = Promise;`.

I mean, I get this warning 100% in my project and I know where it comes from since replacing `return next()` with `next(); return null` in here removes it, but I couldn't get to a minimal repro.